### ISSUE TITLE
Getting a lot of errors using SQL Script

### ISSUE THREAD

**Issue by kuncevic** [duartealexf/sql-ddl-to-json-schema#86]
Hey there first of all thank you for putting an effort crating such useful library. Just recently discovered that.

Here is my question:

I've got a bunch of MS SQL Server tables like that :

```
CREATE TABLE namespace.dbo.MyTable (
	ID int IDENTITY(1,1) NOT NULL,
	FirstName nvarchar(150) COLLATE SQL_Latin1_General_CP1_CI_AS NOT NULL,
	LastName nvarchar(150) COLLATE SQL_Latin1_General_CP1_CI_AS NOT NULL,
	Email nvarchar(150) COLLATE SQL_Latin1_General_CP1_CI_AS NOT NULL,
	HomeAddress nvarchar(250) COLLATE SQL_Latin1_General_CP1_CI_AS NOT NULL,
	City nvarchar(150) COLLATE SQL_Latin1_General_CP1_CI_AS NOT NULL,
	State nvarchar(50) COLLATE SQL_Latin1_General_CP1_CI_AS NOT NULL,
	Country nvarchar(50) COLLATE SQL_Latin1_General_CP1_CI_AS NOT NULL,
	Zip nvarchar(50) COLLATE SQL_Latin1_General_CP1_CI_AS NOT NULL,
	Phone nvarchar(50) COLLATE SQL_Latin1_General_CP1_CI_AS NOT NULL,
	DateOfBirth datetime NOT NULL,
	UserName nvarchar(150) COLLATE SQL_Latin1_General_CP1_CI_AS NOT NULL,
	Password nvarchar(150) COLLATE SQL_Latin1_General_CP1_CI_AS NOT NULL,
	IsPro bit NOT NULL,
	FirstInitDone bit NULL,
	IsLearning bit DEFAULT 0 NOT NULL,
	Active bit NULL,
	ActivationCode uniqueidentifier NULL,
	Gender nvarchar(50) COLLATE SQL_Latin1_General_CP1_CI_AS NULL,
	Status nvarchar(50) COLLATE SQL_Latin1_General_CP1_CI_AS NULL,
	Occupation nvarchar(50) COLLATE SQL_Latin1_General_CP1_CI_AS NULL,
	PartnersOccupation nvarchar(50) COLLATE SQL_Latin1_General_CP1_CI_AS NULL,
	DependantChildren nvarchar(50) COLLATE SQL_Latin1_General_CP1_CI_AS NULL,
	IsUserInfoFilled bit NULL,
	PaidLevel int NULL,
	CONSTRAINT PK_MyTable PRIMARY KEY (ID)
);
```

Passing that as an input

```

const sql = `my_table_string_here`

/**
 * Read on for available options.
 */
const options = {};

/**
 * Get the JSON Schema if you need to modify it...
 */
const jsonSchemaDocuments = parser.feed(sql).toJsonSchemaArray(options);

/**
 * Or explore the compact JSON format...
 */
const compactJsonTablesArray = parser.feed(sql).toCompactJson(parser.results);
```

At first got an error complaining about the namespace that I fixed by removing the namespace

```
Error: Syntax error at line 2 col 23:

  CREATE TABLE namespace.dbo.MyTable (
                        ^
Unexpected S_UNKNOWN token: ".dbo.MyTable (". Instead, I was expecting to see one of the following:

A WS token based on:
    __$ebnf$1 →  ● %WS
    __ →  ● __$ebnf$1
```

Then got into this

```
Error: Syntax error at line 3 col 9:

  	ID int IDENTITY
          ^
Unexpected S_IDENTIFIER_UNQUOTED token: "IDENTITY". Instead, I was expecting to see one of the following:

A WS token based on:
    __$ebnf$1 → __$ebnf$1 ● %WS
    __ →  ● __$ebnf$1


```

I wonder if I am doing something wrong or is it my SQL Script is in wrong format, etc and if so how can I fix that?

p.s: I do not have an SQL Server installed, the SQL Script generated by DBeaver client.

UPDATE: just for test converted my single table MS SQL statement into MySQL using online tool, now getting different sort of errors...., then converted into PostgreSQL and still no luck...

A few moments later.....seems I had to provide MySQL as supported syntax.

Here it is

```
CREATE TABLE PK_MyTable (
	ID int AUTO_INCREMENT NOT NULL,
	FirstName nvarchar(150) NOT NULL,
	LastName nvarchar(150) NOT NULL,
	Email nvarchar(150) NOT NULL,
	HomeAddress nvarchar(250) NOT NULL,
	City nvarchar(150) NOT NULL,
	State nvarchar(50) NOT NULL,
	Country nvarchar(50) NOT NULL,
	Zip nvarchar(50) NOT NULL,
	Phone nvarchar(50) NOT NULL,
	DateOfBirth datetime NOT NULL,
	UserName nvarchar(150) NOT NULL,
	Password nvarchar(150) NOT NULL,
	IsPro bit NOT NULL,
	FirstInitDone bit NULL,
	IsLearning bit DEFAULT 0 NOT NULL,
	Active bit NULL,
	ActivationCode uniqueidentifier NULL,
	Gender nvarchar(50) NULL,
	Status nvarchar(50) NULL,
	Occupation nvarchar(50) NULL,
	PartnersOccupation nvarchar(50) NULL,
	DependantChildren nvarchar(50) NULL,
	IsUserInfoFilled bit NULL,
	PaidLevel int NULL,
	CONSTRAINT PK_MyTable PRIMARY KEY (ID)
);
```

Parser (`const parser = new Parser('mysql');`) errors:

```

Error: Syntax error at line 4 col 12:

  	FirstName nvarchar
             ^
Unexpected S_IDENTIFIER_UNQUOTED token: "nvarchar". Instead, I was expecting to see one of the following:

A WS token based on:
    _$ebnf$1 → _$ebnf$1 ● %WS
```

**Comment by duartealexf** [comment]
Hey @kuncevic, SQL Server syntax is not supported. Only MySQL syntax is.

I'll add support for the `nvarchar` token to be parsed.

**Comment by kuncevic** [comment]
thank you @duartealexf  🔥

I've just noticed that later, just for test changed `nvarchar` into `varchar`, now fighting `uniqueidentifier`.... 🤪 

update: It seems that `uniqueidentifier` should be replaced with `UUID`, so now I understand the source of the poroblems. I will manually verify the data types and convert each one to the corresponding MySQL equivalent. I have no problems with that. 

The online conversion tool misled me! If you happen to know of any reliable ones, please kindly point me in the right direction.

**Comment by duartealexf** [comment]
I actually don't any conversion tool, sorry. Does DBeaver export as MySQL, or have you tried ChatGPT to convert it for you?
Do you still need the `nvarchar` and `uniqueidentifier` ?

**Comment by kuncevic** [comment]
Originally I did google a couple of options to convert MS SQL Script into the MySQL but they didn't do the job properly resulting the unknown types error in `sql-ddl-to-json-schema`. Oh yeah if it does make sense to support `nvarchar` and `uniqueidentifier` in `sql-ddl-to-json-schema` then I guess the answer is yes. However atm I am going trough the manual conversion from MS SQL Script into the MySQL fixing the types so it works for `sql-ddl-to-json-schema`.

**Comment by kuncevic** [comment]
Originally I did google a couple of options to convert MS SQL Script into the MySQL but they didn't do the job properly resulting the unsupported types error in `sql-ddl-to-json-schema`. Oh yeah if it does make sense to support `nvarchar` and `uniqueidentifier` in `sql-ddl-to-json-schema` then I guess the answer is yes. However atm I am going trough the manual conversion from MS SQL Script into the MySQL fixing the types so it works for `sql-ddl-to-json-schema`.

**Comment by kuncevic** [comment]
Used `UUID` insteard of `uniqueidentifier` got that error:


```
Error: Syntax error at line 20 col 20:

      ActivationCode UUID
                     ^
Unexpected K_UUID token: "UUID". Instead, I was expecting to see one of the following:
```

**Comment by duartealexf** [comment]
Hi @kuncevic thank you for your patience.

The datatypes `uuid`, `uniqueidentifier` and `nvarchar` are available as of v4.1.0. 🎉
The mysql version of the `PK_MyTable` is now parseable.

Enjoy.
